Command/DoctrineCommand.php

```diff
@@ -48,7 +48,7 @@ public static function configureMigrations(ContainerInterface $container, Config
             $pathPlaceholderArray = ['kernel.root_dir', 'kernel.cache_dir', 'kernel.logs_dir'];
 
             foreach ($pathPlaceholderArray as $pathPlaceholder) {
-                if (! $container->hasParameter($pathPlaceholder) || ! preg_match('/\%' . $pathPlaceholder . '\%/', $dir)) {
+                if (! $container->hasParameter($pathPlaceholder) || preg_match('/\%' . $pathPlaceholder . '\%/', $dir) === 0) {
```

Comment:
Wouldn't `!== 1` cover the error case too? It is currently not the same logic.

Reply:
Yes… but maybe I should add proper error handling (throw if it is `false`?). Given the value of `$pathPlaceholderArray`, I'd say this is never supposed to crash. Also, should `$pathPlaceholder` go through `preg_quote`? There is a `.` in it.

Reply:
> maybe I should add proper error handling (throw if it is false?)

That would be awesome, but isn't this also a change of the current behaviour? Or do you want to `continue` in the `catch` while still in the 2.x branches?

> Also, should $pathPlaceholder go through preg_quote?

Good point. The values in `pathPlaceholder` are hardcoded, but because code never stays the same forever this could prevent problems.

Reply:
> That would be awesome, but isn't this also a change of the current behaviour? Or do you want to continue in the catch while still in the 2.x branches?

I'm willing to bet it will never throw if nobody touches `$placeholderArray`, so I can't see how that would be a BC-break. I'm proposing this because we don't seem to use an assert library in this package.

:thinking: what would you prefer between that and a call to `assert()`?

Reply:
> thinking what would you prefer between that and a call to assert()?

Nah, I'm fine with `preg_quote`. You convinced me with the `preg_match` check.